modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/PersistentPageMemory.java

```diff
@@ -1835,7 +1861,11 @@ public int maxCheckpointBufferPages() {
     }
 
     private void releaseCheckpointBufferPage(long tmpBufPtr) {
-        checkpointPool.releaseFreePage(tmpBufPtr);
+        int resultCounter = checkpointPool.releaseFreePage(tmpBufPtr);
+
+        if (writeThrottle != null && resultCounter == checkpointPool.pages() / 2) {
```

Comment:
Shouldn't we replace it with `>>> 1`, which is faster?

Reply:
No, JIT'll do it for us